### ISSUE TITLE
[BUG] False positive with load balancers that use a public IP

### ISSUE THREAD

**Issue by BernieWhite** [Azure/PSRule.Rules.Azure#2814]
### Existing rule

Azure.LB.AvailabilityZone

### Description of the issue

`properties.frontendIPConfigurations[*].zones` is only valid with an internal load balancer and not when the load balancer is configured with a public IP.

As a rule the rule is failing with a false positive.

### Reproduction

```bicep
var frontendIPConfigurationsVar = [
  for (frontendIPConfiguration, index) in frontendIPConfigurations: {
    name: frontendIPConfiguration.name
    properties: {
      subnet: contains(frontendIPConfiguration, 'subnetId') && !empty(frontendIPConfiguration.subnetId)
        ? {
            id: frontendIPConfiguration.subnetId
          }
        : null
      publicIPAddress: contains(frontendIPConfiguration, 'publicIPAddressId') && !empty(frontendIPConfiguration.publicIPAddressId)
        ? {
            id: frontendIPConfiguration.publicIPAddressId
          }
        : null
      privateIPAddress: contains(frontendIPConfiguration, 'privateIPAddress') && !empty(frontendIPConfiguration.privateIPAddress)
        ? frontendIPConfiguration.privateIPAddress
        : null
      privateIPAddressVersion: contains(frontendIPConfiguration, 'privateIPAddressVersion')
        ? frontendIPConfiguration.privateIPAddressVersion
        : 'IPv4'
      privateIPAllocationMethod: contains(frontendIPConfiguration, 'subnetId') && !empty(frontendIPConfiguration.subnetId)
        ? (contains(frontendIPConfiguration, 'privateIPAddress') ? 'Static' : 'Dynamic')
        : null
      gatewayLoadBalancer: contains(frontendIPConfiguration, 'gatewayLoadBalancer') && !empty(frontendIPConfiguration.gatewayLoadBalancer)
        ? {
            id: frontendIPConfiguration.gatewayLoadBalancer
          }
        : null
      publicIPPrefix: contains(frontendIPConfiguration, 'publicIPPrefix') && !empty(frontendIPConfiguration.publicIPPrefix)
        ? {
            id: frontendIPConfiguration.publicIPPrefix
          }
        : null
    }
    zones: contains(frontendIPConfiguration, 'subnetId') && !empty(frontendIPConfiguration.subnetId)
    ? [
      '1'
      '2'
      '3'
    ]
    :null
  }
]
```

### Version of PSRule

2.9.0

### Version of PSRule for Azure

1.35.2

### Additional context

https://github.com/arnoldna/bicep-registry-modules/actions/runs/8621385310/job/23630221539
